console/v1/types_console_yaml_sample.go

```diff
@@ -0,0 +1,26 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleYAMLSample is an extension for customizing OpenShift web console YAML samples.
+type ConsoleYAMLSample struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ConsoleYAMLSampleSpec `json:"spec"`
+}
+
+// ConsoleYAMLSampleSpec is the desired YAML sample configuration.
+type ConsoleYAMLSampleSpec struct {
```

Comment:
Looks like we're missing a way to specify the target resource apiVersion and kind

Reply:
`metav1.TypeMeta` contains `apiVersion` and `kind` that should target specific resource

Reply:
i think that field needs a more obvious name (rather than just inlining it) if that's how you're using it, to make it clear it's defining the resourcetype this sample applies to.

Reply:
Definitely, already added a comment in https://github.com/openshift/api/pull/465#discussion_r334489751